Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs

```diff
@@ -0,0 +1,234 @@
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected HttpStatusCode to be an error{reason}, but HttpResponseMessage was <null>.");
+
+            if (success)
+            {
+                Execute.Assertion
+                    .ForCondition(IsClientError() || IsServerError())
+                    .BecauseOf(because, becauseArgs)
+                    .FailWith("Expected HttpStatusCode to be an error{reason}, but found {0}.", Subject.StatusCode);
+            }
+
+            return new AndConstraint<TAssertions>((TAssertions)this);
+        }
+
+        /// <summary>
+        /// Asserts that the <see cref="HttpStatusCode"/> is client error (4xx).
+        /// </summary>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> HaveClientError(string because = "", params object[] becauseArgs)
+        {
+            var success = Execute.Assertion
+                .ForCondition(Subject is not null)
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected HttpStatusCode to be client error (4xx){reason}, but HttpResponseMessage was <null>.");
+
+            if (success)
+            {
+                Execute.Assertion
+                    .ForCondition(IsClientError())
+                    .BecauseOf(because, becauseArgs)
+                    .FailWith("Expected HttpStatusCode to be client error (4xx){reason}, but found {0}.", Subject.StatusCode);
+            }
+
+            return new AndConstraint<TAssertions>((TAssertions)this);
+        }
+
+        /// <summary>
+        /// Asserts that the <see cref="HttpStatusCode"/> is server error (5xx).
+        /// </summary>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> HaveServerError(string because = "", params object[] becauseArgs)
+        {
+            var success = Execute.Assertion
+                .ForCondition(Subject is not null)
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected HttpStatusCode to be server error (5xx){reason}, but HttpResponseMessage was <null>.");
+
+            if (success)
```

Comment:
🔧 You can also do this by chaining those calls together. See https://github.com/fluentassertions/fluentassertions/blob/42f0f8f001cc8f291ae6d3c6efac1bf0da34100f/Src/FluentAssertions/Collections/GenericCollectionAssertions.cs#L115

Reply:
Would you prefer that pattern?

Reply:
With a 🔧 I mean it's a suggestion. Up to you if you still want to do it. I've already approved the PR